tests/validator_test.go

```diff
@@ -329,4 +338,618 @@ var _ = Describe("Template validator", func() {
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			// TODO: Investigate why this test consistently fails without Eventually
+			Eventually(func() error {
+				return apiClient.Create(ctx, vm)
+			}, 180*time.Second).Should(BeNil(), "Failed to create VM")
+		})
+		It("[test_id:5033]: Template with validations, VM without validations", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 2, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:2960] Negative test - Start a VM with machine type violation", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			// set value unfulfilling validation
+			vmi = addDomainResourcesToVMI(vmi, 2, "test", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("test with template optional rules unfulfilled", func() {
+			template = TemplateWithRulesOptional()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("test with template rules unfulfilled", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Should have given error failing to fulfill validations")
```

Comment:
A typo: `ToNot(` -> `To(`. The rules are not fulfilled, so the validation should fail. Strange that the CI passed for this one. Maybe there is a bug in validator.

Reply:
The only rule not fulfilled here is the CPU number, but I see that if it's 0, the validator mutates it to 1, which makes the rule fulfilling
Changed the description to match the test